src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -46,9 +46,10 @@ def __init__(self, host: str, daemon_id: str,
         """
         self.host: str = host
         self.daemon_id = daemon_id
+
         daemon_type = daemon_type or (spec.service_type if spec else None)
         assert daemon_type is not None
-        self.daemon_type: str = daemon_type
+        self.daemon_type: DaemonType = daemon_type  # type: ignore
```

Comment:
this is a bug and needs to be fixed. We can't mix daemon type and service type anymore